test/test_video_player.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:video_player_platform_interface/video_player_platform_interface.dart';
+
+void setupMockVideoPlayerPlatform() {
+  VideoPlayerPlatform.instance = MockVideoPlayerPlatform();
+}
+
+class MockVideoPlayerPlatform extends Fake
```

Comment:
```suggestion
class FakeVideoPlayerPlatform extends Fake
```

That is, let's call this a "fake" instead of a "mock".

Broadly, a mock's method implementations do nothing except record they were called, in a log the test will then inspect. A fake's method implementations do some facsimile of what a real implementation would do, just with all the expensive parts replaced by shortcuts.

This is already more of a fake than a mock, because e.g. the test looks at `isPlaying` rather than counting calls to `play` and `pause`. And that's a good thing — generally fakes are preferable to mocks. They make the tests less coupled to accidental implementation details of the code under test, and better aligned to the code's actual interface and spec.